src/Symfony/Component/EventDispatcher/EventDispatcher.php

```diff
@@ -54,7 +54,7 @@ public function dispatch($event/*, string $eventName = null*/)
 
         if (\is_object($event)) {
             $eventName = $eventName ?? \get_class($event);
-        } elseif (\is_string($event) && (null === $eventName || $eventName instanceof Event)) {
+        } elseif (\is_string($event) && (null === $eventName || $event instanceof ContractsEvent || $eventName instanceof Event)) {
```

Comment:
you mentioned using `\is_object($eventName)`, let's do it, no?

Reply:
I wasn't sure if you wanted people to pass random objects as events. :)